gax/src/main/java/com/google/api/gax/rpc/TransportDescriptor.java

```diff
@@ -40,11 +39,6 @@
 @BetaApi
 public abstract class TransportDescriptor {
```

Comment:
This is not related to this particular change, and this class will be eventually removed, but can you please clarify why are you doing abstract classes with `UnsupportedOperationException` instead of just an interface (this pattern is reused in other places, like for stubs, for example)? It seems like a proper job for interfaces. If any common implementations will be needed, there can be an abstract class added in between.